### PR TITLE
upgrade kotest to 5.2.1

### PR DESCRIPTION
**Pull request by myuwono**

### CODE REVIEW

**Review by sksamuel — APPROVED** [review]

**Comment by myuwono** [comment]
hmm.. the laws dsl in this repo doesn't work anymore due to the way that's being declared with string spec.

File: kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt [hunk 1]
```diff
@@ -2,27 +2,26 @@ package io.kotest.property.arrow.laws
...
+      registerTest(TestName(law.name), disabled = false, config = null) { law.test(this) }
     }
 
-public fun RootScope.testLaws(prefix: String, vararg laws: List<Law>): Unit =
+public suspend fun ContainerScope.testLaws(prefix: String, vararg laws: List<Law>): Unit =
   laws
     .flatMap { list: List<Law> -> list.asIterable() }
     .distinctBy { law: Law -> law.name }
     .forEach { law: Law ->
-      addTest(TestName(prefix, law.name, true), disabled = false, config = null) { law.test(this) }
```
Comment (myuwono) [inline kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt:27]: @i-walker this `addTest` is no longer allowed after spec is instantiated https://github.com/kotest/kotest/pull/2872

Comment (sksamuel) [reply kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt:27]: It never was, it just silently ignored the test. So in 5.2 it actually lets you know it's broken.

Comment (myuwono) [reply kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt:27]: @i-walker I believe under normal circumstances we normally do that within container scope. I tried fixing that here.

Comment (i-walker) [reply kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt:27]: Awesome thank you I didn‘t know the context :)

**Comment by i-walker** [comment]
Thanks @myuwono I‘ll take a look this evening 👍🏾☺️

File: kotest-property-arrow/src/commonTest/kotlin/io/kotest/property/arrow/core/MonoidLawTests.kt [hunk 2]
```diff
@@ -1,19 +1,15 @@
...
 
-class MonoidLawTests : StringSpec({
-  "Boolean, Int, String obeys MonoidLaws" {
-    testLaws(
-      MonoidLaws.laws(Monoid.boolean(), Arb.boolean()),
-      MonoidLaws.laws(Monoid.int(), Arb.int()),
-      MonoidLaws.laws(Monoid.string(), Arb.string())
-    )
-  }
+class MonoidLawTests : FunSpec({
```
Comment (i-walker) [inline kotest-property-arrow/src/commonTest/kotlin/io/kotest/property/arrow/core/MonoidLawTests.kt:11]: can't we use StringSpec here?

File: kotest-property-arrow-optics/src/commonTest/kotlin/io/kotest/property/arrow/optics/TraversalTests.kt [hunk 3]
```diff
@@ -14,44 +14,43 @@ import io.kotest.property.arrow.core.functionAToB
...
 
-class TraversalTests : StringSpec({
-  "Traversal Laws for Optional" {
-    testLaws(
-      TraversalLaws.laws(
-        traversal = Optional.listHead<Int>(),
-        aGen = Arb.list(Arb.int()),
-        bGen = Arb.int(),
-        funcGen = Arb.functionAToB(Arb.int()),
+class TraversalTests : FunSpec({
```
Comment (i-walker) [inline kotest-property-arrow-optics/src/commonTest/kotlin/io/kotest/property/arrow/optics/TraversalTests.kt:17]: same here

**Review by i-walker — APPROVED** [review]
Left a few nits, thanks @myuwono 👍🏾

**Pull request merged**